tests/aws/services/stepfunctions/v2/mocking/test_base_callbacks.py

```diff
@@ -0,0 +1,149 @@
+import json
+
+import pytest
+from localstack_snapshot.snapshots.transformer import JsonpathTransformer
+
+from localstack import config
+from localstack.testing.aws.util import is_aws_cloud
+from localstack.testing.pytest import markers
+from localstack.testing.pytest.stepfunctions.utils import (
+    create_and_record_execution,
+    create_and_record_mocked_execution,
+    create_state_machine_with_iam_role,
+)
+from localstack.utils.strings import short_uid
+from tests.aws.services.stepfunctions.mocked_responses.mocked_response_loader import (
+    MockedResponseLoader,
+)
+from tests.aws.services.stepfunctions.templates.base.base_templates import BaseTemplate
+from tests.aws.services.stepfunctions.templates.callbacks.callback_templates import (
+    CallbackTemplates,
+)
+
+
+@markers.snapshot.skip_snapshot_verify(
+    paths=[
+        "$..SdkHttpMetadata",
+        "$..SdkResponseMetadata",
+        "$..ExecutedVersion",
+        "$..RedriveCount",
+        "$..RedriveStatus",
+        "$..RedriveStatusReason",
+        # In an effort to comply with SFN Local's lack of handling of sync operations,
+        # we are unable to produce valid TaskSubmittedEventDetails output field, which
+        # must include the provided mocked response in the output:
+        "$..events..taskSubmittedEventDetails.output",
+    ]
+)
+class TestBaseScenarios:
```

Comment:
question: Given that the mocking feature is only available for StepFunctions Local, how are we doing parity testing here?
Currently, we validate the test against real AWS and then have adjusted logic for recording mocked execution, which should match the real AWS behavior, right. It might be worth adding a clarifying comment at the top of the test file and/or somewhere in the StepFunction service documentation.


Just a thought: It's probably not worth investing into snapshot tests against StepFunction Local if it's planned to become discontinued.

Reply:
I will clarify this point in the documentation. But in short, we validate our support to mocking is in parity with AWS SFN, and we infer the API behavior of SFN Local wrt. mocked services from testing (this). SFN Local otherwise has a set of parity issues when it comes to evaluation. This method seems the most future-proof for LocalStack; any parity issues it exposes in today's users’ mock environments I would not consider it as a limitation of our implementation.